frontend/src/views/Storage/Storage.tsx

```diff
@@ -124,31 +138,44 @@ export default function Storage() {
             </ButtonGov>
           </ButtonGovGroup>
         </div>
-        <StorageModal
-          key={isDialogOpen.toString()}
-          initialModalData={storageFormData}
-          storageList={storageList}
-          handleDialogClose={handleDialogClose}
-          handleSubmit={handleSubmit}
-          isOpen={isDialogOpen}
-          onOpenChange={handleDialogClose}
-          isDismissable
-        />
+        {isDialogOpen && (
+          <StorageModal
+            initialModalData={
+              rowEditIndex !== undefined
+                ? state.nmpFile.years[0].ManureStorageSystems![rowEditIndex]
+                : undefined
+            }
+            rowEditIndex={rowEditIndex}
+            unassignedManures={unassignedManures}
+            handleDialogClose={handleDialogClose}
+            isOpen={isDialogOpen}
+          />
+        )}
         <TabsMaterial
           activeTab={2}
           tabLabel={['Add Animals', 'Manure & Imports', 'Storage', 'Nutrient Analysis']}
         />
       </>
       <DataGrid
         sx={{ ...customTableStyle, marginTop: '1.25rem' }}
-        rows={storageList}
+        rows={state.nmpFile.years[0].ManureStorageSystems || []}
         columns={columnsAnimalManure}
-        getRowId={(row: any) => row.index}
+        getRowId={() => crypto.randomUUID()}
         disableRowSelectionOnClick
         disableColumnMenu
         hideFooterPagination
         hideFooter
       />
+      <div>
```

Comment:
I know this is dumb and ugly but I'd rather unblock the other StorageModal tickets and just tell Josh it'll visually get fixed later.